### DISCUSSION TITLE
评论通知官网文档没有写单纯的webhook这种方式吗？ || Comment to inform that the official website document does not write a simple webhook method?

### DISCUSSION THREAD

**Discussion by axiaoxin** [walinejs/waline#2655] [Q&A] [answered]
我看官网文档里通知方式没有单独的webhook，都是第三方的，是不支持直接请求一个配置的url吗？但我看讨论里面有人问webhook相关问题。请问有文档吗？谢谢

---
I see that there is no separate webhook for notification methods in the official website documentation. They are all third-party. Doesn’t it support directly requesting a configured URL? But I saw someone in the discussion asked webhook related questions. Is there any documentation? Thank you

**Comment by axiaoxin** [comment]
另外，我看到有飞书群聊机器人的webhook，请问是否支持企业微信的群聊机器人webhook，两个差不多的

---
In addition, I saw a webhook for the Feishu group chat robot. Is it possible to support the corporate WeChat group chat robot webhook? There are two similar

**Comment by lizheming** [comment, marked as answer]
https://waline.js.org/reference/server/env.html#%E9%AB%98%E7%BA%A7%E9%85%8D%E7%BD%AE

**Comment by mapxn** [comment]
这里有一个相关的[issue](https://github.com/walinejs/waline/issues/2661)，我打算根据飞书的webhook添加一个企微群机器人的通知，测试好后提交PR。
本地代码已经基本改好了。在本地测试我执行`pnpm server:dev` 后，根据提示登录vercel账户，自动在vercel部署一个测试项目，启动了localhost:9000 端口，发现本地修改的代码貌似没有生效，看起来引用的还是cdn的js文件。
<img width="830" alt="image" src="https://github.com/user-attachments/assets/666faa76-41a9-4493-96c4-dad09750114e" />


我不是写JS的，有佬能帮忙指导下怎么调试本地`/packages/server/src/service/notify.js`代码么？

---
There is a related issue here (https://github.com/walinejs/waline/issues/2661). I plan to add a notification for the Qiwei group robot based on Feishu's webhook, and submit a PR after testing it.
The local code has basically been changed. In the local test, after I executed `pnpm server:dev`, I logged in to the vercel account according to the prompts, automatically deployed a test project in vercel, and started the localhost:9000 port. I found that the locally modified code did not seem to take effect, and it seemed that the reference was still from the cdn. js file.
<img width="830" alt="image" src="https://github.com/user-attachments/assets/666faa76-41a9-4493-96c4-dad09750114e" />


I am not a JS writer. Can anyone help guide me on how to debug the local `/packages/server/src/service/notify.js` code?

**Reply by lizheming** [reply]
前端页面目前已经是配置到本地服务端服务了，只要配置对应的环境变量开启相关的功能即可。

---
The front-end page is currently configured to the local server service. You only need to configure the corresponding environment variables to enable related functions.

**Reply by mapxn** [reply]
> 前端页面目前已经是配置到本地服务端服务了，只要配置对应的环境变量开启相关的功能即可。
> 
> The front-end page is currently configured to the local server service. You only need to configure the corresponding environment variables to enable related functions.

@lizheming  感谢指点，我的步骤如下：

测试环境：本地 / Github Codespaces
1. fork项目
2. git clone到本地
3. 复制 .env.example 到 .env，修改LEAN_ID，LEAN_KEY，LEAN_MASTER_KEY这三个变量（使用国际版 / leancloud.app）
4. pnpm install
5. pnpm server:dev
6. 按提示登录Vercel，如下：
```
> No existing credentials found. Please log in:
? Log in to Vercel Continue with GitHub
> Success! GitHub authentication complete for myemail@outlook.com
? Set up and develop “/workspaces/waline/example”? yes
? Which scope should contain your project? Myscope
? Link to existing project? no
? What’s your project’s name? example
? In which directory is your code located? ./
🔗  Linked to oneboy/example (created .vercel)
> Ready! Available at http://localhost:9090
```

7. 浏览器访问`http://localhost:9090`，正常进入界面，简单输入邮箱，昵称，网址以及评论内容，点击submit，如下图：
<img width="832" alt="image" src="https://github.com/user-attachments/assets/baa60e08-7ffb-4715-bef4-16459b8d059c" />

8. 确认leancloud的配置有效（LEAN_ID，LEAN_KEY，LEAN_MASTER_KEY），若使用生产环境的项目配置，可以加载出生产环境的评论，只是无法提交评论。

9. 报错日志如下：

```shell
Vercel CLI 39.2.6
> Ready! Available at http://localhost:9090
BadRequestError: Incorrect parameter format
    at Object.throw (/workspaces/waline/node_modules/.pnpm/koa@2.15.3/node_modules/koa/lib/context.js:97:11)
    at /workspaces/waline/node_modules/.pnpm/think-payload@1.4.0/node_modules/think-payload/index.js:56:16
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at cors (/workspaces/waline/node_modules/.pnpm/@koa+cors@5.0.0/node_modules/@koa/cors/index.js:109:16)
    at /workspaces/waline/packages/server/src/middleware/version.js:5:3
    at /workspaces/waline/packages/server/src/middleware/prefix-warning.js:27:3
```

MAC电脑本地测试，Github Codespaces两个环境都是这样。

已经调试两天了，还没有头绪，是我哪里不会用吗？

**Reply by lizheming** [reply]
之前排查过这个问题，结论是 Vercel CLI 有点问题，还没来得及给他们反馈。可以直接通过 `node ./packages/server/vanilla.js` 启动吧。

---
I have checked this problem before and concluded that there is something wrong with Vercel CLI and I haven’t had time to give them feedback yet. You can start it directly through `node ./packages/server/vanilla.js`.
